src/composables/Project.js

```diff
@@ -69,3 +87,68 @@ export function deleteProjectById(projectId) {
 export function deleteAllProjects() {
   localStorage.setItem(PROJECT_STORAGE_KEY, '{}');
 }
+
+/**
+ * Update remote origin, fetch and checkout the default branch.
+ * @param {Project} project - Project to update.
+ * @return {Promise<void>} Promise with nothing on success otherwise an error.
+ */
+export function updateGitProject(project) {
+  const dir = `/${project.id}`;
+
+  return git.addRemote({
+    fs,
+    dir,
+    remote: 'origin',
+    force: true,
+    url: project.git.repository,
+  })
+    .then(() => git.fetch({
+      fs,
+      http,
+      dir,
+      corsProxy: 'https://cors.isomorphic-git.org',
+    })).then(() => git.listBranches({ fs, dir, remote: 'origin' }))
+    .then((branches) => {
+      const ref = branches.filter((branche) => branche !== 'HEAD').find(() => true);
+      return git.checkout({
+        fs,
+        dir,
+        ref,
+        force: true,
+      });
+    });
+}
+
+/**
+ * Retrieve list of project files name.
+ * @param {String} projectId - Id of project.
+ * @return {Promise<FileInformation[]>} Promise with file names array on success,
+ * otherwise error.
+ */
+export function getProjectFiles(projectId) {
+  const project = getProjectById(projectId);
+  return git.listFiles({ fs, dir: `/${project.id}` })
+    .then((files) => files.map((file) => new FileInformation({ path: file })));
+}
+
+/**
+ * Get file content.
+ * @param {String} projectId - Id of project.
+ * @param {FileInformation} fileInformation - Object that contain file path.
+ * @return {Promise<FileInput>} Promise with file content on success otherwise error.
+ */
+export function readProjectFile(projectId, fileInformation) {
+  const project = getProjectById(projectId);
+  const dir = `/${project.id}`;
+
+  return git.resolveRef({ fs, dir, ref: 'main' })
+    .then((commitOid) => git.readBlob({
+      fs,
+      dir,
+      oid: commitOid,
+      filepath: fileInformation.path,
+    }))
+    .then(({ blob }) => Buffer.from(blob).toString('utf8'))
+    .then((content) => new FileInput({ path: fileInformation.path, content }));
```

Comment:
As per above, use async/await and explicitely return the newly created FileInput.

Reply:
Not agree for the async/await

Reply:
See with you, improve readability and remove promise chaining.
Done